core/src/main/java/org/apache/carbondata/core/carbon/querystatistics/DriverQueryStatisticsRecorder.java

```diff
@@ -78,106 +82,142 @@ public synchronized void recordStatisticsForDriver(QueryStatistic statistic, Str
-          String line2 = "|" + printLine(" ", len1) + "+" + printLine("-", len2) + "+" +
-              printLine(" ", len3) + "+" + printLine("-", len4) + "+";
-          // table header
-          tableInfo.append(line).append("\n");
-          tableInfo.append("|" + printLine(" ", (len1 - "Module".length())) + "Module" + "|" +
-              printLine(" ", (len2 - "Operation Step".length())) + "Operation Step" + "|" +
-              printLine(" ", (len3 + len4 + 1 - "Query Cost".length())) +
-              "Query Cost" + "|" + "\n");
-          // driver part
-          tableInfo.append(line).append("\n");
-          tableInfo.append("|" + printLine(" ", len1) + "|" +
-              printLine(" ", (len2 - "SQL parse".length())) + "SQL parse" + "|" +
-              printLine(" ", len3) + "|" +
-              printLine(" ", (len4 - sql_parse_time.length())) + sql_parse_time + "|" + "\n");
-          tableInfo.append(line2).append("\n");
-          tableInfo.append("|" +printLine(" ", (len1 - "Driver".length())) + "Driver" + "|" +
-              printLine(" ", (len2 - "Load meta data".length())) + "Load meta data" + "|" +
-              printLine(" ", (len3 - driver_part_time.length())) + driver_part_time + "|" +
-              printLine(" ", (len4 - load_meta_time.length())) +
-              load_meta_time + "|" + "\n");
-          tableInfo.append(line2).append("\n");
-          tableInfo.append("|" +
-              printLine(" ", (len1 - "Part".length())) + "Part" + "|" +
-              printLine(" ", (len2 - "Block allocation".length())) +
-              "Block allocation" + "|" +
-              printLine(" ", len3) + "|" +
-              printLine(" ", (len4 - block_allocation_time.length())) +
-              block_allocation_time + "|" + "\n");
-          tableInfo.append(line2).append("\n");
-          tableInfo.append("|" +
-              printLine(" ", len1) + "|" +
-              printLine(" ", (len2 - "Block identification".length())) +
-              "Block identification" + "|" +
-              printLine(" ", len3) + "|" +
-              printLine(" ", (len4 - block_identification_time.length())) +
-              block_identification_time + "|" + "\n");
-          tableInfo.append(line).append("\n");
-
-          // once the statistics be printed, remove it from the map
-          queryStatisticsMap.remove(key);
-          // show query statistic as "query id" + "table"
-          return "Print query statistic for query id: " + key + "\n" + tableInfo.toString();
+  public String collectDriverStatistics(String key) {
+    String sql_parse_time = "";
+    String load_meta_time = "";
+    String block_allocation_time = "";
+    String block_identification_time = "";
+    Double driver_part_time_tmp = 0.0;
+    Double driver_part_time_tmp2 = 0.0;
+    String splitChar = " ";
+    try {
+      // get statistic time from the QueryStatistic
+      for (QueryStatistic statistic : queryStatisticsMap.get(key)) {
```

Comment:
null possibility here also, pass entry.getValue() instead, once you iterate over entrySet() in the calling method